ArduPlane/mode_LoiterAltQLand.cpp

```diff
@@ -38,7 +38,9 @@ void ModeLoiterAltQLand::switch_qland()
 {
     ftype dist;
     if ((!plane.current_loc.get_alt_distance(plane.next_WP_loc, dist) || is_negative(dist)) && plane.nav_controller->reached_loiter_target()) {
-        plane.set_mode(plane.mode_qland, ModeReason::LOITER_ALT_REACHED_QLAND);
+        const Location loiter_wp = plane.next_WP_loc;
+        plane.set_mode(plane.mode_qrtl, ModeReason::LOITER_ALT_REACHED_QLAND);
+        plane.set_next_WP(loiter_wp);
```

Comment:
how is altitude handled?

Reply:
It's not, which is one of many problems with this PR.